app/models/station.rb

```diff
@@ -1,3 +1,71 @@
 class Station < ActiveRecord::Base
-  validates :name, :dock_count, :city, :installation_date, presence: true
+  belongs_to :city
+  validates :name,
+            :dock_count,
+            :city_id,
+            :installation_date,
+            presence: true
+
+  def self.record(station_information)
+    self.find_or_create_by(name:       station_information[:name],
+                           dock_count: station_information[:dock_count],
+                           city_id: find_city_id(station_information[:city_name]),
+                           installation_date: station_information[:installation_date]
+                          )
+  end
+
+  def self.record_update(station, station_information)
```

Comment:
this is the new update method being called on in the route. the reason we can't use the built in "update" method is that our station model requires a city_id in order to validate. we can't expect our user to know what the city id is, so we need something that translates the name that they enter into the city_id in order for the station to be valid.